owncloudApp/src/main/java/com/owncloud/android/presentation/files/createfolder/CreateFolderDialogFragment.kt

```diff
@@ -71,7 +72,23 @@ class CreateFolderDialogFragment : DialogFragment() {
             }
             .setNegativeButton(android.R.string.cancel, null)
             .setTitle(R.string.uploader_info_dirname)
-        val alertDialog: Dialog = builder.create()
+        val alertDialog = builder.create()
+
+        // When the dialog is shown for the first time we disable the button
+        // There's a listener listening to change in the input field ->
+        // first  it trims
+        // when field has some text -> button enabled
+        // when field is empty -> button disabled
+        alertDialog.setOnShowListener{
+            val button = alertDialog.getButton(AlertDialog.BUTTON_POSITIVE)
+            button.isEnabled = false
+            inputText.addTextChangedListener {
+                if (it != null) {
+                    button.isEnabled = it.trim().isNotEmpty()
+                }
+            }
+        }
```

Comment:
You caught the concept! 🚀 But I suggest you some changes to make your code look cleaner:

```suggestion
        alertDialog.setOnShowListener {
            val okButton = alertDialog.getButton(AlertDialog.BUTTON_POSITIVE)
            okButton.isEnabled = false
        }
        inputText.doOnTextChanged { text, _, _, _ ->
            val okButton = alertDialog.getButton(AlertDialog.BUTTON_POSITIVE)
            okButton.isEnabled = !text.isNullOrBlank())
        }
```
The first callback is useful for disabling the button the first time the dialog is shown as you documented above. The second callback is a Kotlin extension that will allow us to create a listener easily as you can see (more info: https://stackoverflow.com/a/73978429). In this case, I would use `isNullOrBlank` so that we also avoid folder with names made by only blank spaces .

And yes, in this case we need to define the `okButton` variable inside both callbacks because outside of them, the button does not exist 👍 .